dcicutils/aws_utils.py

```diff
@@ -0,0 +1,261 @@
+""" Functions to support general aws operations
+
+    Function names should generally follow a convention to include a short string
+    to indicate the service that they are designed to operate with eg. 's3' for AWS s3.
+"""
+import boto3
+import logging
+import mimetypes
+
+from botocore.exceptions import ClientError
+# from .exceptions import ()
+from .s3_utils import s3Utils
+
+###########################
+# Config
+###########################
+logging.basicConfig()
+logger = logging.getLogger(__name__)
+
+
+def s3_bucket_head(*, bucket_name, s3=None):
+    """ Gets head info for a bucket if it exists
+
+    :param bucket_name: name of the bucket - string
+    :param s3: AWS s3 client
+    :return: dict: head response or None
+    """
+    try:
+        s3 = s3 or s3Utils().s3
```

Comment:
I would not do this this way as it will implicitly pull in a bunch of other things when calling `s3Utils()` - I would just do `boto3.client('s3')`. This comment applies to many functions throughout here.